### PR TITLE
implemented admin global and counterparty management

### PR DESCRIPTION
**Pull request by DFSko**
implemented admin global and counterparty management

resolves happy-travel/agent-app-project#62
resolves happy-travel/agent-app-project#63

### CODE REVIEW

File: Api/AdministratorServices/CounterpartyManagementService.cs [hunk 1]
```diff
@@ -54,15 +54,27 @@ select new
...
+                    formula.AgencyId,
+                    formula.AgentId
+                } into formulas
+                from markupFormula in formulas.DefaultIfEmpty()
                 select new
                 {
                     Counterparty = cp,
-                    Country = c
+                    Country = c,
+                    MarkupFormula = markupFormula == null ? null : markupFormula.DisplayFormula
```
Comment (mingazhev) [inline Api/AdministratorServices/CounterpartyManagementService.cs:74]: Is it the shortest possible way to write?

Comment (DFSko) [reply Api/AdministratorServices/CounterpartyManagementService.cs:74]: EF core can't translate null conditional operator to SQL

File: HappyTravel.Edo.Common/Enums/Markup/MarkupPolicyEventOperationType.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace HappyTravel.Edo.Common.Enums.Markup
+{
+    public enum MarkupPolicyEventOperationType
+    {
+        Created,
```
[the remark this answers is not in the document]
Comment (DFSko) [reply HappyTravel.Edo.Common/Enums/Markup/MarkupPolicyEventOperationType.cs]: Fixed

File: Api/Services/Markups/DisplayedMarkupFormulaService.cs [hunk 3]
```diff
@@ -89,6 +89,45 @@ public async Task<Result> Update(int agencyId)
...
+            }
+
+            await _context.SaveChangesAsync();
+            return Result.Success();
+        }
+
+
+        public async Task<Result> UpdateGlobalFormula()
+        {
+            // TODO: implement saving global markup formulas
```
[the remark this answers is not in the document]
Comment (DFSko) [reply Api/Services/Markups/DisplayedMarkupFormulaService.cs:125]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**